operator/common/common.go

```diff
@@ -40,9 +40,17 @@ func CreateResourceName(maxLen int, minSubstrLen int, str ...string) string {
 				str[i] = str[i][:len(str[i])-cut]
 			}
 		} else {
-			return strings.ToLower(newStr)
+			return sanitizeResourceNameString(newStr)
```

Comment:
since we now change _ to - should we update the crd-yaml info on **metadata.name**? In all those files we specify that all names must follow k8s spec which is no more true....

Reply:
maybe adding "all _ will be substituted by - " ?

Reply:
the requirement of resource names to adhere to the k8s spec is still true - this PR just avoids that we attempt to create a resource which does not follow these requirements